src/app/modules/register/components/register/register.component.ts

```diff
@@ -0,0 +1,134 @@
+    RouterLink,
+    NgIf,
+  ],
+  viewProviders: [provideIcons({ jamGithub, jamGoogle })],
+})
+export class RegisterComponent {
+  passwordMatch(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      if (!control.parent || !control.parent.get('password')) return null;
+
+      const targetControl = control.parent.get('password')!;
+
+      const result = targetControl.value === control.value ? null : { passwordMatch: true };
+      return result;
+    };
+  }
+
+  usernameValidator(): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> => {
+      return control.valueChanges.pipe(
+        debounceTime(300),
+        switchMap(value => this.dbService.findProfileByUsername(value)),
+        map(user => (user ? { usernameTaken: true } : null)),
+        first(),
+      );
+    };
+  }
+
+  emailFormControl = new FormControl('', [Validators.required, Validators.email]);
+  nameFormControl = new FormControl('', [Validators.required, Validators.minLength(2)]);
+  surnameFormControl = new FormControl('', [Validators.required, Validators.minLength(2)]);
+  passwordFormControl = new FormControl('', [
+    Validators.required,
+    Validators.pattern(/^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*\W)(?=.*\S)(?!.* ).{8,}$/),
+  ]);
+  passwordConfirmFormControl = new FormControl('', [Validators.required, this.passwordMatch()]);
+  usernameFormControl = new FormControl('', [Validators.required, Validators.minLength(3)], [this.usernameValidator()]);
+  registerForm: FormGroup;
+
+  userID$ = this.store.select(selectUserID);
+  userProfile$ = this.store.select(selectUserProfile);
+  userAuthenticated$ = this.store.select(selectIsAuthenticated);
+
+  constructor(
+    private store: Store<UserState>,
+    private formB: FormBuilder,
+    private dbService: DbService,
+  ) {
+    this.registerForm = this.formB.group({
+      email: this.emailFormControl,
+      name: this.nameFormControl,
+      surname: this.surnameFormControl,
+      password: this.passwordFormControl,
+      passwordConfirm: this.passwordConfirmFormControl,
+      username: this.usernameFormControl,
+    });
+  }
+
+  signUpGoogle(): void {
+    this.store.dispatch(registerGoogle());
+  }
+
+  signUpGit(): void {
+    this.store.dispatch(registerGit());
+  }
+
+  signUpUser(): void {
+    if (
+      !this.emailFormControl.value ||
+      !this.passwordFormControl.value ||
+      !this.nameFormControl.value ||
+      !this.surnameFormControl.value ||
+      !this.usernameFormControl.value
+    )
+      return;
+    const userProfileInfo: IUser = {
+      username: this.usernameFormControl.value,
+      name: this.nameFormControl.value,
+      surname: this.surnameFormControl.value,
+    };
+    this.store.dispatch(
+      register({
+        userProfile: userProfileInfo,
+        password: this.passwordFormControl.value,
+        email: this.emailFormControl.value,
+      }),
+    );
+  }
+
+  ngOnInit(): void {}
```

Comment:
Unused Init function. Unless there is future use for it, then it is completely fine👍

Reply:
Removed, thanks bro!